internal/wire/wire_test.go

```diff
@@ -320,3 +372,238 @@ func TestPersistView(t *testing.T) {
+
+			// Delete the persisted collection to clean up.
+			f = func(c *mgo.Collection) error {
+				return c.DropCollection()
+			}
+
+			if err := db.ExecuteMGO(tests.Context, "testcollection", f); err != nil {
+				t.Fatalf("\t%s\tShould be able to drop the output collection : %s", tests.Failed, err)
+			}
+			t.Logf("\t%s\tShould be able to drop the output collection.", tests.Success)
+		}
+	}
+}
+
+// TestExecuteNameFail tests that the correct result is returned when
+// an invalid view name is provided.
+func TestExecuteNameFail(t *testing.T) {
+	tests.ResetLog()
+	defer tests.DisplayLog()
+
+	db, err := db.NewMGO(tests.Context, tests.TestSession)
+	if err != nil {
+		t.Fatalf("\t%s\tShould be able to get a Mongo session : %v", tests.Failed, err)
+	}
+	defer db.CloseMGO(tests.Context)
+
+	opts := make(map[string]interface{})
+	opts["database_name"] = cfg.MustString("MONGO_DB")
+	opts["username"] = cfg.MustString("MONGO_USER")
+	opts["password"] = cfg.MustString("MONGO_PASS")
+	store, err := cayley.NewGraph("mongo", cfg.MustString("MONGO_HOST"), opts)
+	if err != nil {
+		t.Fatalf("\t%s\tShould be able to get a Cayley handle : %v", tests.Failed, err)
+	}
+
+	// -----------------------------------------------------------
+	// Generate the view.
+
+	t.Log("Given the need to catch an invalid view name.")
+	{
+		t.Log("\tWhen using the view, relationship, and item fixtures.")
+		{
+
+			// Form the view parameters.
+			viewParams := wire.ViewParams{
+				ViewName: viewPrefix + "this view name does not exist",
+				ItemKey:  "ITEST_80aa936a-f618-4234-a7be-df59a14cf8de",
+			}
+
+			// Generate the view.
+			result, err := wire.Execute(tests.Context, db, store, &viewParams)
+			if err == nil {
+				t.Fatalf("\t%s\tShould return an error for an invalid view name: %s", tests.Failed, err)
+			}
+			t.Logf("\t%s\tShould return an error for an invalid view name", tests.Success)
+
+			// Check the resulting items.
+			errDoc, ok := result.Results.(bson.M)
+			if !ok || len(errDoc["error"].(string)) == 0 {
+				t.Fatalf("\t%s\tShould return a single error document : %s", tests.Failed, err)
+			}
+			t.Logf("\t%s\tShould return a single error document.", tests.Success)
+		}
+	}
+}
+
+// TestExecuteTypeFail tests that the correct result is returned when
+// an invalid start type is defined in view metadata.
+func TestExecuteTypeFail(t *testing.T) {
+	tests.ResetLog()
+	defer tests.DisplayLog()
+
+	db, err := db.NewMGO(tests.Context, tests.TestSession)
+	if err != nil {
+		t.Fatalf("\t%s\tShould be able to get a Mongo session : %v", tests.Failed, err)
+	}
+	defer db.CloseMGO(tests.Context)
+
+	opts := make(map[string]interface{})
+	opts["database_name"] = cfg.MustString("MONGO_DB")
+	opts["username"] = cfg.MustString("MONGO_USER")
+	opts["password"] = cfg.MustString("MONGO_PASS")
+	store, err := cayley.NewGraph("mongo", cfg.MustString("MONGO_HOST"), opts)
+	if err != nil {
```

Comment:
Possible to extract the database setup into a `setup` function as was done for xenia? Ref: https://github.com/coralproject/shelf/pull/192/commits/adae938946d25a22f954a2c2a978c483e8bd4f4f